src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -128,16 +141,14 @@ public IAfterCallSpecifiedConfiguration Throws(Func<IFakeObjectCall, Exception>
             public IAfterCallSpecifiedWithOutAndRefParametersConfiguration ReturnsLazily(Func<IFakeObjectCall, TMember> valueProducer)
             {
                 Guard.AgainstNull(valueProducer, nameof(valueProducer));
-
+                this.ParentConfiguration.EnsureRuleAdded();
                 this.ParentConfiguration.RuleBeingBuilt.UseApplicator(x => x.SetReturnValue(valueProducer(x)));
                 return this.ParentConfiguration;
             }
 
             public IReturnValueConfiguration<TMember> Invokes(Action<IFakeObjectCall> action)
             {
-                Guard.AgainstNull(action, nameof(action));
-
-                this.ParentConfiguration.RuleBeingBuilt.Actions.Add(action);
+                this.ParentConfiguration.Invokes(action);

```

Comment:
This was doing the same as the parent implementation from `RuleBuilder`, so I thought it might as well call the parent implementation directly.